### PR TITLE
Type parameterized lj function tests

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

Restructure tests to properly test LJFunctor and LJFunctorAVX by means of type parameterized tests.

- [x] Globals tests
- [x] NoGlobals tests
- [x] VS tests
- [x] Readable test names
- [x] Bugfixes in LJFunctorAVX discovered by new tests

## Related Pull Requests

- merges into #423 

## ~Resolved Issues~

# How Has This Been Tested?

- [x] Merged existing tests like `testAoSFunctorNoGlobalsNoN3` and `testAoSFunctorNoGlobalsN3` into one test implementation that generates the former two.
- [x] Generated new tests out of existing ones but with different functors.
- [x] Skip tests that use not implemented functions.
- [x] fixed all new tests.

### CODE REVIEW

File: tests/testAutopas/tests/pairwiseFunctors/LJFunctorTest.cpp [hunk 1]
```diff
@@ -1,667 +1,7 @@
 /**
```
Comment (SteffenSeckler) [inline tests/testAutopas/tests/pairwiseFunctors/LJFunctorTest.cpp]: delete this file

File: tests/testAutopas/tests/pairwiseFunctors/LJFunctorTestGlobals.cpp [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+  using TestType = LJFunctorTestGlobals<FuncType>;
+
+  for (size_t additionalDummyParticles = 0; additionalDummyParticles < 30; additionalDummyParticles += 5) {
+    for (bool duplicatedCalculation : {false, true}) {
+      // the own functor can only be called for inner or outside pairs! (if two particles lie in one cell they can be
+      // either both inside the process or neither of them is)
+      for (typename TestType::where_type where : {TestType::inside, TestType::boundary, TestType::outside}) {
+        for (bool newton3 : {false, true}) {
+          for (bool cellWiseOwnedState : {false, true}) {
+            for (uint64_t numParticleReplicas : {1, 2}) {
```
Comment (SteffenSeckler) [inline tests/testAutopas/tests/pairwiseFunctors/LJFunctorTestGlobals.cpp]: I'd like it if these for loops were also all generated, i.e., that there is a generated test for each different value of these parameters...

File: tests/testAutopas/tests/pairwiseFunctors/LJFunctorTestGlobals.cpp [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+TYPED_TEST_P(LJFunctorTestGlobals, testSoAFunctorGlobalsPair) {
+  using FuncType = TypeParam;
+  using TestType = LJFunctorTestGlobals<FuncType>;
+
+  for (bool duplicatedCalculation : {false, true}) {
+    for (typename TestType::where_type where : {TestType::inside, TestType::boundary, TestType::outside}) {
+      for (bool newton3 : {false, true}) {
+        for (bool cellWiseOwnedState : {false, true}) {
+          for (uint64_t numParticleReplicas : {1, 2}) {
+            this->testSoAGlobals(where, newton3, duplicatedCalculation, TestType::pair, 0, cellWiseOwnedState,
```
Comment (SteffenSeckler) [inline tests/testAutopas/tests/pairwiseFunctors/LJFunctorTestGlobals.cpp]: same here

File: tests/testAutopas/tests/pairwiseFunctors/LJFunctorTestVs.cpp [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+REGISTER_TYPED_TEST_SUITE_P(LJFunctorTestVs, testSetPropertiesVSPPLSoA, testSetPropertiesVSPPLAoS);
+
+using MyTypes = ::testing::Types<
+    // LJFunctor<mixing> VS LJFunctor<not mixing>
+    std::tuple<LJFunShiftMixGlob, LJFunShiftNoMixGlob>
+#ifdef __AVX__
+    ,
+    // LJFunctorAVX<mixing> VS LJFunctorAVX<not mixing>
```
Comment (SteffenSeckler) [inline tests/testAutopas/tests/pairwiseFunctors/LJFunctorTestVs.cpp]: document this test a bit, why can you compare `mixing` vs `not mixing` (same type), ...

File: tests/testAutopas/tests/pairwiseFunctors/LJFunctorTestGlobals.cpp (hunk 2 above)
Comment (FG-TUM) [reply tests/testAutopas/tests/pairwiseFunctors/LJFunctorTestGlobals.cpp]: Not easily possible:
https://stackoverflow.com/questions/8507385/google-test-is-there-a-way-to-combine-a-test-which-is-both-type-parameterized-a

**Pull request merged**